src/gui/qgsattributeform.cpp

```diff
@@ -1446,6 +1473,22 @@ void QgsAttributeForm::init()
           layout->addWidget( label, row, column++ );
           layout->addWidget( widgetInfo.widget, row, column++ );
         }
+
+        // Alias DD overrides
+        if ( widgDef->type() == QgsAttributeEditorElement::AttributeEditorType::AeTypeField )
+        {
+          const QgsAttributeEditorField *fieldElement { static_cast<QgsAttributeEditorField *>( widgDef ) };
+          const QString fieldName { mLayer->fields().at( fieldElement->idx() ).name() };
+          if ( mLayer->editFormConfig().dataDefinedFieldProperties( fieldName ).hasProperty( QgsEditFormConfig::DataDefinedProperty::Alias ) )
+          {
+            const QgsProperty property { mLayer->editFormConfig().dataDefinedFieldProperties( fieldName ).property( QgsEditFormConfig::DataDefinedProperty::Alias ) };
+            const QString labelExpression { property.asExpression() };
+            if ( ! labelExpression.isEmpty() && property.isActive() )
+            {
+              mExpressionLabels[ label ] = QgsExpression( labelExpression );
```

Comment:
Why not mExpressionProperties and just evaluate the property directly?

Reply:
Because it would invalidate the whole purpose of this PR: make the labels "dynamic" in the form context. 
Ok, maybe a more complex approach would be to check if the expression is "dynamic" here and pre-evaluate it if it is not. But leaving alone the complexity (or even the possibility) of determining if the expression is "dynamic" or not, I don't think it would add any speed gain: we would still need to check if the expression is dynamic whenever a form value changes.

Reply:
I tried your advice but it doesn't work, the reason is that the expression values are cached inside the expression evaluation code, I need to create a new expression every time.
The issue comes from the (wrong) assumption that if an expression function takes a constant parameter it will return always the same result, but expressions like `current_value( 'my_field_name')` will return different values depending on the form context (the current status of the edited feature in the form at evaluation time).

Reply:
Make sure to tag the `current_value` function as not static (i.e. `QgsExpressionFunction::isStatic` returns `false`)

Reply:
Thanks @m-kuhn that was I was looking for. I'll try that.

Reply:
@m-kuhn that was the right solution! Now everything works as expected.